### DISCUSSION TITLE
Practice Mode

### DISCUSSION THREAD

**Discussion by Panthr75** [ppy/osu#17949] [Ideas] [answered]
Implementation to add a specific mode to practice a particular beatmap.

**Comment by bdach** [comment, marked as answer]
did you read any of the existing multiple discussion threads on this?

https://github.com/ppy/osu/discussions/16723
https://github.com/ppy/osu/discussions/14629
https://github.com/ppy/osu/discussions/13079
https://github.com/ppy/osu/discussions/13076
https://github.com/ppy/osu/discussions/13013, esp. https://github.com/ppy/osu/discussions/13013#discussioncomment-2467752

**Reply by Panthr75** [reply]
To be honest no, because this was just a feature I added in a private fork and thought it would be good in base game

**Comment by Panthr75** [comment]
This feature is actually useless in retrospective because of osu playtest mode

**Reply by bdach** [reply]
i don't know about that, it still has a chance of being useful/merged. just not as a single slider that uses map completion percentage.

**Reply by Panthr75** [reply]
This would fit better as an additional `Player` mode than a mod

**Comment by Panthr75** [comment]
There seems to be a problem with "seeking" where the music/replay still continues as seeking occurs

https://user-images.githubusercontent.com/29237603/164942759-938bb2b3-9220-45e3-b47a-223eeb1ab8b3.mp4

**Reply by frenzibyte** [reply]
That looks to be out of scope from this thread..? please open a separate issue/discussion thread for it (and try searching first).

**Reply by Panthr75** [reply]
If you want to seek to a certain position in the song, dealing with this would be a problem and is very relevant

**Reply by frenzibyte** [reply]
It is still out of scope from the concept of "practice mode" as proposed here, we can't deal with multiple "issues" packed in one thread (I'm also not even sure how it's a problem with implementation of "Practice Mode" other than introducing unnecessary noise for a short time).

**Reply by Panthr75** [reply]
Someone brought it up in [another thread](https://github.com/ppy/osu/discussions/13013#discussioncomment-791257) for seeking to be implemented in this "practice mode". Here I have created a basic example on why this could be a problem.

https://user-images.githubusercontent.com/29237603/164943154-454642bf-9628-426b-81fc-f36ed6706485.mp4

The player does not see the track and cannot respond to notes until a couple of seconds after the "seek point"

This doesn't necessarily mean that it needs to be fixed in this issue, but will just be a known problem as the logic for seeking is used for both this example practice mode and replay mode.

It doesn't make sense to create a new system that's already implemented.
